### PR TITLE
Lint assets too

### PR DESCRIPTION
**Pull request by XhmikosR**
This is totally WIP. I'm just trying to get the most important things out.

I'll rebase when the other PRs are merged.

Best reviewed with the non-whitespace diff: <https://github.com/emedvedev/slackin-extended/pull/25/files?w=1>

### CODE REVIEW

**Comment by XhmikosR** [comment]
@emedvedev I've disabled most checks for assets so that we tackle them in batches. Can you have a look at the current errors? Remember you can push to my branch if needed :)

File: assets/client.js [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 var button = body.querySelector('button')
 
 // remove loading state
 button.classList.remove('loading')
 
 // capture submit
-function submitForm (ev){
-  ev && ev.preventDefault()
+function submitForm(ev) {
+  if (ev) ev.preventDefault()
```
Comment (XhmikosR) [inline assets/client.js]: I'm not sure if this is needed. When can the event not be passed?

**Comment by XhmikosR** [comment]
@emedvedev can you please disable the PR comments on lgtm.com dashboard?

**Comment by emedvedev** [comment]
Done, sorry :)

**Comment by emedvedev** [comment]
I'll take a look at the errors a bit later today, should be able to merge by EOD.

**Comment by emedvedev** [comment]
Not entirely sure why the `Set Node.js version` fails in the GitHub pipeline.

**Comment by XhmikosR** [comment]
It's an issue with GH Actions which I assume it will be solved automatically when people at GitHub notice the issue, if they haven't already.

File: assets/superagent.js [hunk 2]
```diff
@@ -682,7 +682,6 @@
  */
 
     function Response(req, options) {
-      options = options || {}
```
Comment (XhmikosR) [inline assets/superagent.js]: Are we sure this wasn't supposed to be used? Like `this.options`?

Comment (emedvedev) [reply assets/superagent.js]: It's not passed downstream and there's no reference to it anywhere else, so I think it should be safe.

**Comment by emedvedev** [comment]
GitHub tests pass now, and the LGTM issues in assets are fixed as well, so we can merge this. I'll create a separate PR for the LGTM issues outside of assets.

**Comment by XhmikosR** [comment]
I actually want to reduce the suppressed rules too eventually :)

**Comment by emedvedev** [comment]
Ah! Sorry for merging too soon then, haha. Feel free to open another PR for tackling the suppressed rules :)

**Pull request merged**